tests/validate_regexes.rs

```diff
@@ -0,0 +1,72 @@
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+use std::fs;
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+struct Example {
+    #[serde(rename(deserialize = "Valid"))]
+    valid: Vec<String>,
+    #[serde(rename(deserialize = "Invalid"), default)]
+    invalid: Vec<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+struct TestCase {
+    #[serde(rename(deserialize = "Name"))]
+    name: String,
+    #[serde(rename(deserialize = "Regex"))]
+    regex: String,
+    plural_name: bool,
+    #[serde(rename(deserialize = "Description"))]
+    description: Option<String>,
+    #[serde(rename(deserialize = "Rarity"))]
+    rarity: f64,
+    #[serde(rename(deserialize = "URL"))]
+    url: Option<String>,
+    #[serde(rename(deserialize = "Tags"))]
+    tags: Vec<String>,
+    #[serde(rename(deserialize = "Examples"), default)]
+    examples: Example,
+}
+
+#[test]
+fn validate_regex_examples() {
+    let path = format!("{}/src/data/regex.json", env!("CARGO_MANIFEST_DIR"));
+    let data: String = fs::read_to_string(path).unwrap();
+
+    let test_cases: Vec<TestCase> = serde_json::from_str(&data).unwrap();
+    for test_case in &test_cases {
+        let re = Regex::new(&test_case.regex);
+        match re {
+            Ok(re) => {
+                for example in &test_case.examples.valid {
+                    let matched = re.is_match(example);
+                    if !matched {
+                        println!(
+                            "INVALID MATCH for {}\nexample: {}\nregex: {}",
```

Comment:
This is the case where it should have matched the example, so its not an `INVALID MATCH`.

Replace it with something like `DIDN'T MATCH` or `SHOULD MATCH`

Reply:
fixed